src/player.cpp

```diff
@@ -3174,6 +3169,12 @@ void Player::postRemoveNotification(Thing* thing, const Cylinder* newParent, int
 	}
 
 	if (const Item* item = thing->getItem()) {
+		if (item->isSupply()) {
+			if (Player* player = item->getHoldingPlayer()) {
```

Comment:
if postRemoveNotification is called, this means player is holding the item (or was), so this check is redundant

Reply:
this check is needed, without it, the player will receive an update supply when moving the item on the ground or throwing it to the ground

Reply:
oh right, fair point. I forgot that postRemove works for every single remove, but wouldnt that mean you can just use newParent and compare it with this?
```
if (newParent == this)
```

Reply:
I prefer this form, it is more readable for me